pyflakes/checker.py

```diff
@@ -870,6 +922,41 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        # Complain if there are duplicate keys with different values
+        # If they have the same value it's not going to cause potentially
+        # unexpected behaviour so we'll not complain.
+        keys = [
+            convert_to_value(key) for key in node.keys
+        ]
+        for key in set(keys):
+            if keys.count(key) > 1:
```

Comment:
Feels like this would be cleaner (eg. less cyclomatic complexity) if written as

```python
if keys.count(key) == 0:
    continue

candidate_values = []
# [..]
```